appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/model/ExtensibleImpl.java

```diff
@@ -83,16 +90,33 @@ public static void merge(Extension from, Extensible<?> to, boolean override) {
             to.setExtensions(new LinkedHashMap<>());
         }
         if (from.name() != null && !from.name().isEmpty()) {
-            Object value = mergeProperty(to.getExtensions().get(from.name()), convertExtensionValue(from.value()),
-                    override);
+            Object value = mergeProperty(to.getExtensions().get(from.name()), 
+                    convertExtensionValue(from.value(), from.parseValue()), override);
             to.getExtensions().put(from.name(), value);
         }
     }
 
-    public static Object convertExtensionValue(String value) {
+    public static Object convertExtensionValue(String value, boolean parseValue) {
         if (value == null) {
             return null;
         }
+        if (parseValue) {
+            if (value.startsWith("{") || value.startsWith("[")) { // JSON object + array
+                try {
+                    return new ObjectMapper().readTree(value);
+                } catch (Exception e) {
+                    return value;
```

Comment:
Logging the exception on debug-level might help the developers that didn't expect the result to be string. Actually, if they explicitly set `parseValue=true`, they likely do not expect string back at all. So it's likely worth a warning.